Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -111,19 +114,17 @@ export function getSonarQubeEndpoint(): SonarQubeEndpoint {
 }
 
 // Upload a build summary with links to available SonarQube dashboards for further analysis details.
-export function uploadSonarQubeBuildSummary(sqBuildFolder: string): void {
-    // Save and upload build summary
-    // Looks like: "[Detailed SonarQube report >](https://mySQserver:9000/dashboard/index/foo "foo Dashboard")"
-    var buildSummaryContents:string = createSonarQubeBuildSummary(sqBuildFolder);
-
-    var buildSummaryFilePath = saveSonarQubeBuildSummary(buildSummaryContents);
-
-    tl.debug('Uploading build summary from ' + buildSummaryFilePath);
-
-    tl.command('task.addattachment', {
-        'type': 'Distributedtask.Core.Summary',
-        'name': tl.loc('sqAnalysis_BuildSummaryTitle')
-    }, buildSummaryFilePath);
+export function uploadSonarQubeBuildSummaryIfEnabled(sqBuildFolder:string):Q.Promise<void> {
```

Comment:
uploadSonarQubeBuildSummary, previously a synchronous method, now constructs and resolves a promise chain. This is because of the need to wait for the SonarQube server to finish analysing the build and give the quality gate status.